src/renderers/shared/utils/__tests__/ReactErrorUtils-test.js

```diff
@@ -132,36 +135,56 @@ describe('ReactErrorUtils', () => {
     it(`does not return nested errors (${environment})`, () => {
       const err1 = new Error();
       let err2;
-      const err3 = ReactErrorUtils.invokeGuardedCallback(
+      ReactErrorUtils.invokeGuardedCallback(
         'foo',
         function() {
-          err2 = ReactErrorUtils.invokeGuardedCallback(
+          ReactErrorUtils.invokeGuardedCallback(
             'foo',
             function() {
               throw err1;
             },
             null,
           );
+          err2 = ReactErrorUtils.clearCaughtError();
         },
         null,
       );
+      // Returns null because inner error was already captured
+      expect(ReactErrorUtils.hasCaughtError()).toBe(false);
 
-      expect(err3).toBe(null); // Returns null because inner error was already captured
       expect(err2).toBe(err1);
     });
 
+    if (!__DEV__) {
+      // jsdom doesn't handle this properly, but Chrome and Firefox should. Test
+      // this with a fixture.
```

Comment:
We run this whole suite in both prod and dev mode. https://github.com/acdlite/react/blob/155648e26724083132b5fb3a2172c7d288533e08/src/renderers/shared/utils/__tests__/ReactErrorUtils-test.js#L22-L45

The dev version of `invokeGuardedCallback` is only enabled in dev if it passes a feature test. jsdom happens to fail that feature test, because it doesn't handle `throw null`. So I added a flag to ReactDOMFeatureFlags that forces the use of the dev version. https://github.com/acdlite/react/blob/155648e26724083132b5fb3a2172c7d288533e08/src/renderers/shared/utils/ReactErrorUtils.js#L187-L191

For Chrome and Firefox, I think we'll have to rely on a manual browser fixture, unless we start running automated browser tests.